server/item/fire_charge.go

```diff
@@ -0,0 +1,34 @@
+package item
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/sound"
+	"github.com/go-gl/mathgl/mgl64"
+	"math/rand"
+	"time"
+)
+
+// FireCharge is an item that can be used to place fire when used on a block, or shot from a dispenser to create a small
+// fireball.
+type FireCharge struct{}
+
+// EncodeItem ...
+func (f FireCharge) EncodeItem() (name string, meta int16) {
+	return "minecraft:fire_charge", 0
+}
+
+// UseOnBlock ...
+func (f FireCharge) UseOnBlock(pos cube.Pos, face cube.Face, _ mgl64.Vec3, w *world.World, _ User, ctx *UseContext) bool {
+	ctx.SubtractFromCount(1)
```

Comment:
Although this does work correctly, I would encourage you to only call `ctx.SubtractFromCount(1)` in the case where you `return true` to avoid confusion in the future.